.github/workflows/cmake-par-script.yml

```diff
@@ -0,0 +1,230 @@
+name: hdf5 dev parallel ctest script runs
+
+# Triggers the workflow on a call from another workflow
+on:
+  workflow_call:
+    inputs:
+      snap_name:
+        description: 'The name in the source tarballs'
+        type: string
+        required: false
+        default: hdfsrc
+      file_base:
+        description: "The common base name of the source tarballs"
+        required: true
+        type: string
+      use_environ:
+        description: 'Environment to locate files'
+        type: string
+        required: true
+        default: snapshots
+      build_mode:
+        description: "release vs. debug build"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  CMake_build_parallel_openmpi:
+    name: "Parallel OpenMPI GCC-${{ inputs.build_mode }}"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install CMake Dependencies (OpenMPI)
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build graphviz curl
+          sudo apt install libssl3 libssl-dev libcurl4 libcurl4-openssl-dev
+          sudo apt install gcc-12 g++-12 gfortran-12
+          sudo apt install libaec0 libaec-dev
+          sudo apt-get install libopenmpi-dev
```

Comment:
I'm sure it's just from copy pasting, but we really don't need pretty much any of these dependencies other than libopenmpi-dev